### PR TITLE
Feature uarm gcode

### PR DESCRIPTION
**Pull request by renskekuip1**

### CODE REVIEW

File: code/headers/robot_arm.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+namespace r2d2::robot_arm {
+    class robot_arm_c {
+    protected:
+        r2d2::hardware_usart_c usart_bus;
+
+    public:
+        /**
+         * Robot arm constructor, needs an usart bus for communication
+         */
+        robot_arm_c(r2d2::hardware_usart_c usart_bus);
```
Comment (Kaaserne) [inline code/headers/robot_arm.hpp]: Most of the functions can take the parameters by `const&`

**Review by Kaaserne — CHANGES_REQUESTED** [review]
I think the virtual methods can be pure virtual methods. For e.g. `virtual void init() = 0;`. If this is the case ,please add a virtual destructor, if not, then you can leave it as is.

File: code/headers/uarm_gcode_generator_c.hpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ namespace r2d2::robot_arm {
          * @return char* to buffer
```
Comment (itzandroidtab) [inline code/headers/uarm_gcode_generator_c.hpp:22]: Why is this file called uarm_gcode_generator```_c```.hpp why not just uarm_gcode_generator.hpp

File: code/main.cpp [hunk 3]
```diff
@@ -1,16 +1,12 @@
-#include <hwlib.hpp>
+#include "dof4_diy_c.hpp"
```
Comment (itzandroidtab) [inline code/main.cpp]: this is a library so it should be with ```<>```

the includes below this one as well

Comment (itzandroidtab) [reply code/main.cpp]: and the same with the file name why the _c?

File: code/headers/robot_arm_interface.hpp [hunk 4]
```diff
@@ -1,30 +1,27 @@
 #pragma once
 
-#include "vector3_c.hpp"
+#include "vector3.hpp"
```
Comment (Kaaserne) [inline code/headers/robot_arm_interface.hpp:3]: All .hpp's have to be included with `<>`

**Comment by itzandroidtab** [comment]
Why would you add a delete? There is probably something differently going wrong

**Review by Kaaserne — APPROVED** [review]

**Pull request merged**